src/components/StatsPanel/StatsPanel.tsx

```diff
@@ -0,0 +1,34 @@
+import React from 'react';
+
+import { useStatsPanel } from './StatsPanel.hook';
+import {
+  CardContainer,
+  StatItem,
+  StatIcon,
+  StatText,
+  StatValue,
+  IconWrapper,
+} from './StatsPanel.styles';
+
+import type { StatsPanelProps } from './StatsPanel.types';
+
+/** StatsPanel component */
+export const StatsPanel: React.FC<StatsPanelProps> = ({ totalTests, passed, failed, waiting }) => {
+  const stats = useStatsPanel({ totalTests, passed, failed, waiting });
+
+  return (
+    <CardContainer>
+      {stats.map((stat) => (
+        <StatItem key={stat.label}>
```

Comment:
Using stat.label as the React key may lead to unstable keys if labels change or are not unique. Keys should be stable and unique (e.g., an id).

**Sugerencia:**
```typescript
Provide a stable id in StatItemData (e.g. 'id' or use the record key), and use it as key: <StatItem key={stat.id}>. If stat keys are known, pass them via useStatsPanel so that each item has a stable id.
```